rest_framework_json_api/filters/queryvalidation.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+from rest_framework.exceptions import ValidationError
+from rest_framework.filters import BaseFilterBackend
+
+
+class QueryValidationFilter(BaseFilterBackend):
+    """
+    A backend filter that performs strict validation of query parameters for
+    jsonapi spec conformance and raises a 400 error if non-conforming usage is
+    found.
+
+    If you want to add some additional non-standard query parameters,
+    simply override `.query_regex` adding the new parameters but, "with the additional
+    requirement that they MUST contain contain at least one non a-z character (U+0061 to U+007A).
```

Comment:
instead of repeating parts of the specification I prefer we only reference the jsonapi spec by link here. If someone wants to break the spec it is really up to them anyway.